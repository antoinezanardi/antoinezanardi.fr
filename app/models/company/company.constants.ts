import type { CompanyName, Company } from "~/models/company/company.types";

const COMPANY_NAMES = [
  "SoBook",
  "OhMyCode",
  "Daveo",
] as const;

const COMPANIES: Record<CompanyName, Company> = Object.freeze({
  OhMyCode: {
    name: "OhMyCode",
    url: "https://www.sobook.fr/oh-my-code/",
    image: "omc-logo.png",
  },
  SoBook: {
    name: "SoBook",
    url: "https://www.sobook.fr/oh-my-code/",
    image: "sobook-logo.png",
  },
  Daveo: {
    name: "Daveo",
    url: "https://www.daveo.fr/",
    image: "daveo-logo.png",
  },
});

export {
  COMPANY_NAMES,
  COMPANIES,
};