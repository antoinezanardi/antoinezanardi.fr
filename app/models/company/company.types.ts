import type { TupleToUnion } from "type-fest";

import type { COMPANY_NAMES } from "~/models/company/company.constants";

type CompanyName = TupleToUnion<typeof COMPANY_NAMES>;

type Company = {
  name: CompanyName;
  image: string;
  url: string;
};

export type { Company, CompanyName };