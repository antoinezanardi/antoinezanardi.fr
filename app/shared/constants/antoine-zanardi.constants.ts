const ANTOINE_ZANARDI_FULL_NAME = "Antoine ZANARDI";

const ANTOINE_ZANARDI_BIRTH_DATE = "1996-04-14";

const ANTOINE_ZANARDI_GITHUB_URL = "https://github.com/antoinezanardi/";

const ANTOINE_ZANARDI_GITHUB_AVATAR_URL = "https://avatars.githubusercontent.com/u/11405981?v=4";

const ANTOINE_ZANARDI_LINKEDIN_URL = "https://www.linkedin.com/in/antoinezanardi/";

export {
  ANTOINE_ZANARDI_FULL_NAME,
  ANTOINE_ZANARDI_BIRTH_DATE,
  ANTOINE_ZANARDI_GITHUB_URL,
  ANTOINE_ZANARDI_GITHUB_AVATAR_URL,
  ANTOINE_ZANARDI_LINKEDIN_URL,
};