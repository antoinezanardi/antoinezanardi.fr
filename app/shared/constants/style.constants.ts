const PRIMARY_COLOR = "#12671D";

const BLACK_COLOR = "#000000";

const WHITE_COLOR = "#FFFFFF";

export {
  PRIMARY_COLOR,
  BLACK_COLOR,
  WHITE_COLOR,
};