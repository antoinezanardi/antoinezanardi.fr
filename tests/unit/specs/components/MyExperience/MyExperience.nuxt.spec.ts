import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import type ProfessionalExperienceCard from "~/components/MyExperience/ProfessionalExperienceCard/ProfessionalExperienceCard.vue";
import { MyExperience } from "#components";
import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";
import { COMPANIES } from "~/models/company/company.constants";
import type { ProfessionalExperience } from "~/models/professional-experience/professional-experience.types";

describe("My Experience Component", () => {
  let wrapper: ReturnType<typeof mount<typeof MyExperience>>;

  async function mountMyExperienceComponent(options: ComponentMountingOptions<typeof MyExperience> = {}):
  Promise<ReturnType<typeof mount<typeof MyExperience>>> {
    return mountSuspendedComponent(MyExperience, { ...options });
  }

  beforeEach(async() => {
    wrapper = await mountMyExperienceComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Experiences", () => {
    it("should render 6 experiences when rendered.", () => {
      const experiences = wrapper.findAllComponents<typeof ProfessionalExperienceCard>(".professional-experience-card");

      expect(experiences).toHaveLength(6);
    });

    it("should render fullstack developer experience card at Daveo when rendered.", () => {
      const experiences = wrapper.findAllComponents<typeof ProfessionalExperienceCard>(".professional-experience-card");
      const expectedExperience: ProfessionalExperience = {
        company: COMPANIES.Daveo,
        job: {
          name: "MyExperience.fullStackDeveloperConsultant",
          startedAt: new Date("2022-04-01"),
          description: [
            "MyExperience.iJoinedDaveo",
            "MyExperience.whatIsConsulting",
            "MyExperience.myMissionsAsConsultant",
            "MyExperience.cloudProductCertifications",
            "MyExperience.conferenceAndWorkshops",
          ],
        },
      };

      expect(experiences[0].props("professionalExperience")).toStrictEqual<ProfessionalExperience>(expectedExperience);
    });

    it("should render it engineer at OhMyCode when rendered.", () => {
      const experiences = wrapper.findAllComponents<typeof ProfessionalExperienceCard>(".professional-experience-card");
      const expectedExperience: ProfessionalExperience = {
        company: COMPANIES.OhMyCode,
        job: {
          name: "MyExperience.itR&DEngineer",
          startedAt: new Date("2019-09-01"),
          finishedAt: new Date("2022-03-31"),
          description: [
            "MyExperience.firstPermanentContract",
            "MyExperience.asProjectManager",
            "MyExperience.softSkills",
            "MyExperience.expertiseAndTechnicalSkills",
          ],
        },
      };

      expect(experiences[1].props("professionalExperience")).toStrictEqual<ProfessionalExperience>(expectedExperience);
    });

    it("should render second intern at OhMyCode when rendered.", () => {
      const experiences = wrapper.findAllComponents<typeof ProfessionalExperienceCard>(".professional-experience-card");
      const expectedExperience: ProfessionalExperience = {
        company: COMPANIES.OhMyCode,
        job: {
          name: "MyExperience.internFullStackDeveloper",
          startedAt: new Date("2018-09-01"),
          finishedAt: new Date("2019-08-30"),
          description: [
            "MyExperience.distributionProject",
            "MyExperience.thisFinalInternship",
          ],
        },
      };

      expect(experiences[2].props("professionalExperience")).toStrictEqual<ProfessionalExperience>(expectedExperience);
    });

    it("should render first intern at OhMyCode when rendered.", () => {
      const experiences = wrapper.findAllComponents<typeof ProfessionalExperienceCard>(".professional-experience-card");
      const expectedExperience: ProfessionalExperience = {
        company: COMPANIES.OhMyCode,
        job: {
          name: "MyExperience.internFullStackDeveloper",
          startedAt: new Date("2016-09-01"),
          finishedAt: new Date("2017-04-30"),
          description: [
            "MyExperience.firstStepsInOhMyCode",
            "MyExperience.allFunctionalitiesInOnApp",
          ],
        },
      };

      expect(experiences[3].props("professionalExperience")).toStrictEqual<ProfessionalExperience>(expectedExperience);
    });

    it("should render freelance developer at SoBook when rendered.", () => {
      const experiences = wrapper.findAllComponents<typeof ProfessionalExperienceCard>(".professional-experience-card");
      const expectedExperience: ProfessionalExperience = {
        company: COMPANIES.SoBook,
        job: {
          name: "MyExperience.freelanceFullStackDeveloper",
          startedAt: new Date("2016-02-01"),
          finishedAt: new Date("2016-08-30"),
          description: [
            "MyExperience.reiterateAsFreelance",
            "MyExperience.moreResponsibilities",
          ],
        },
      };

      expect(experiences[4].props("professionalExperience")).toStrictEqual<ProfessionalExperience>(expectedExperience);
    });

    it("should render first intern at SoBook when rendered.", () => {
      const experiences = wrapper.findAllComponents<typeof ProfessionalExperienceCard>(".professional-experience-card");
      const expectedExperience: ProfessionalExperience = {
        company: COMPANIES.SoBook,
        job: {
          name: "MyExperience.internFullStackDeveloper",
          startedAt: new Date("2015-07-01"),
          finishedAt: new Date("2015-12-31"),
          description: ["MyExperience.firstInternship"],
        },
      };

      expect(experiences[5].props("professionalExperience")).toStrictEqual<ProfessionalExperience>(expectedExperience);
    });
  });
});