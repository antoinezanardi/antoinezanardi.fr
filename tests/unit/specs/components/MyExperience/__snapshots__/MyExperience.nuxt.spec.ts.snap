// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`My Experience Component > should match snapshot when rendered. 1`] = `
"<div id="experience" aria-labelledby="experience-title" class="section" role="region">
  <div class="cc-experience container">
    <section-title-stub icon="fa fa-briefcase" title="ExpÃ©rience professionnelle" iconcolor="#4D9344" id="experience-title"></section-title-stub>
    <professional-experience-card-stub professionalexperience="[object Object]" aria-label="1 - MyExperience.fullStackDeveloperConsultant" class="professional-experience-card" role="region"></professional-experience-card-stub>
    <professional-experience-card-stub professionalexperience="[object Object]" aria-label="2 - MyExperience.itR&amp;DEngineer" class="professional-experience-card" role="region"></professional-experience-card-stub>
    <professional-experience-card-stub professionalexperience="[object Object]" aria-label="3 - MyExperience.internFullStackDeveloper" class="professional-experience-card" role="region"></professional-experience-card-stub>
    <professional-experience-card-stub professionalexperience="[object Object]" aria-label="4 - MyExperience.internFullStackDeveloper" class="professional-experience-card" role="region"></professional-experience-card-stub>
    <professional-experience-card-stub professionalexperience="[object Object]" aria-label="5 - MyExperience.freelanceFullStackDeveloper" class="professional-experience-card" role="region"></professional-experience-card-stub>
    <professional-experience-card-stub professionalexperience="[object Object]" aria-label="6 - MyExperience.internFullStackDeveloper" class="professional-experience-card" role="region"></professional-experience-card-stub>
  </div>
</div>"
`;
